src/mon/OSDMonitor.cc

```diff
@@ -1646,6 +1649,8 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
 
   utime_t grace = orig_grace;
   double my_grace = 0, peer_grace = 0;
+  if (propose_now)
+    *propose_now = false;
```

Comment:
@liewegas  Actually check_failure() will also be called by prepare_failure(), that is why I try to make this argument optional.